src/core/drive/cache_stores/disk_store.ts

```diff
@@ -0,0 +1,54 @@
+import { CacheStore } from "../cache_store"
+import { PageSnapshot } from "../page_snapshot"
+
+export class DiskStore extends CacheStore {
+  _version = "v1"
+  storage!: Cache
+
+  constructor() {
+    super()
+    this.initialize()
+  }
+
+  async initialize() {
+    this.storage = await caches.open(`turbo-${this.version}`)
+  }
+
+  async has(location: URL) {
+    return (await this.storage.match(location)) !== undefined
+  }
+
+  async get(location: URL) {
+    const response = await this.storage.match(location)
+    if (response && response.ok) {
+      const html = await response.text()
+      return PageSnapshot.fromHTMLString(html)
+    }
+  }
+
+  async put(location: URL, snapshot: PageSnapshot) {
+    const response = new Response(snapshot.html, {
+      status: 200,
+      statusText: "OK",
+      headers: {
+        "Content-Type": "text/html",
+      },
+    })
+    await this.storage.put(location, response)
+    return snapshot
+  }
+
+  async clear() {
+    const keys = await this.storage.keys()
+    await Promise.all(keys.map((key) => this.storage.delete(key)))
```

Comment:
I wonder if there's a point where it's quicker to delete the cache itself, rather than the individual items. Perhaps it's very unlikely to come up in any case. But I wondered if that's something you looked at?

Reply:
The cache API only has a method to delete individual keys https://developer.mozilla.org/en-US/docs/Web/API/Cache

Reply:
You can delete the `Cache` instance though, instead of deleting the individual items that it contains: https://developer.mozilla.org/en-US/docs/Web/API/CacheStorage/delete

Reply:
Ah, I missed that 👍

I think deleting the keys is ok, though. Don't think there will be a performance penalty, and because the operations are async there's a small chance of race condition if we delete the cache and reopen it again when there are other operations inflight.